File: FrontEnd/app_loteria/lib/models/Sucursal.dart
class Sucursal {
  late final int sucursalId;
  final int? municipioId;
  final String nombre;
  final String? direccion;

  Sucursal({
    required this.sucursalId,
    this.municipioId,
    required this.nombre,
    this.direccion,
  });

  factory Sucursal.fromJson(Map<String, dynamic> json) {
    return Sucursal(
      sucursalId: json['SucursalId'] as int,
      municipioId: json['MunicipioId'] as int?,
      nombre: json['Nombre'] as String,
      direccion: json['Direccion'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "SucursalId": sucursalId,
      "MunicipioId": municipioId,
      "Nombre": nombre,
      "Direccion": direccion,
    };
  }
}

File: Backend/Numerito.API/Numerito.API/Data/Entities/Sucursal.cs
namespace Numerito.API.Data.Entities;

public partial class Sucursal
{
    public int SucursalId { get; set; }

    public int? MunicipioId { get; set; }

    public string? Nombre { get; set; }

    public string? Direccion { get; set; }

    public int UsuarioCreacion { get; set; }

    public DateTime FechaCreacion { get; set; }

    public int? UsuarioModificacion { get; set; }

    public DateTime? FechaModificacion { get; set; }

    public bool? Estado { get; set; }

    public virtual Municipio? Municipio { get; set; }

    public virtual ICollection<Usuario> Usuarios { get; set; } = new List<Usuario>();
}
